### PR TITLE
[REVIEW] - Fix #206: resolved bug in printing empty dataframe

### LINKED ISSUE #206: Printing an empty dataframe throws an error
## Reporting a bug

- [X] I am using the latest version of PyGDF ~~from conda or~~ built from `master`.
- [X] I have included the following environment details:
      Linux Distro: Ubuntu 16.04, Linux Kernel: Linux 4.15.0-29-generic x86_64 , GPU Model: TITAN V
- [ ] I have included the following version information for:
      Arrow, CUDA,Numpy, Pandas, Python
- [X] I have included below a minimal working reproducer (if you are unsure how
 to write one see http://matthewrocklin.com/blog/work/2018/02/28/minimal-bug-reports).

Example:
```
from pygdf.dataframe import DataFrame
gdf = DataFrame()
print(gdf)
```
Output:
```
---------------------------------------------------------------------------
IndexError                                Traceback (most recent call last)
<ipython-input-58-168e83139bd3> in <module>()
      1 from pygdf.dataframe import DataFrame
      2 gdf = DataFrame()
----> 3 print(gdf)

/conda/envs/gdf/lib/python3.6/site-packages/pygdf-0.1.0a2+315.ge37f495-py3.6.egg/pygdf/dataframe.py in __str__(self)
    262         nrows = settings.formatting.get('nrows') or 10
    263         ncols = settings.formatting.get('ncols') or 8
--> 264         return self.to_string(nrows=nrows, ncols=ncols)
    265 
    266     def __repr__(self):

/conda/envs/gdf/lib/python3.6/site-packages/pygdf-0.1.0a2+315.ge37f495-py3.6.egg/pygdf/dataframe.py in to_string(self, nrows, ncols)
    249         cols = OrderedDict()
    250         use_cols = list(self.columns[:ncols - 1])
--> 251         use_cols.append(self.columns[-1])
    252 
    253         for h in use_cols:

IndexError: tuple index out of range
```

### PR DESCRIPTION
**Pull request by AK-ayush**
This PR fix the existing bug in printing empty dataframe as mentioned in [issue](https://github.com/rapidsai/cudf/issues/206).   
 
There are code changes in three files:
-  **cudf/dataframe.py** 
- **cudf/formatting.py** 
- **cudf/tests/test_dataframe.py**: unit tests are written in it.

**Issues:** First, in `to_string()` method of `cudf/dataframe.py`, `ncols` is not being calculated correctly and second, in same method, it is trying to get the last element of an empty list.

Let me know, what all things are needed more.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -11,7 +11,7 @@ __pycache__
 *.pytest_cache
 build
 htmlcov
-
+.idea
```
Comment (harrism) [inline .gitignore]: Is this change needed? If not, please revert it.

File: cudf/tests/test_dataframe.py [hunk 2]
```diff
@@ -356,6 +356,34 @@ def test_dataframe_to_string_wide():
     assert got.split() == expect.split()
 
 
+def test_dataframe_empty_to_string():
+    # Test for printing empty dataframe
+    df = DataFrame()
+    got = df.to_string()
+    print(got)
+    expect = '''
```
Comment (harrism) [inline cudf/tests/test_dataframe.py]: Please correct the indentation.
```suggestion
    expect = "Empty DataFrame\nColumns: []\nIndex: []\n"
```

File: cudf/tests/test_dataframe.py [hunk 3]
```diff
@@ -356,6 +356,34 @@ def test_dataframe_to_string_wide():
...
+    # values should match despite whitespace difference
+    assert got.split() == expect.split()
+
+
+def test_dataframe_emptycolumns_to_string():
+    # Test for printing dataframe having empty columns
+    df = DataFrame({'a': [], 'b': []})
+    got = df.to_string()
+    print(got)
+    expect = '''
```
Comment (harrism) [inline cudf/tests/test_dataframe.py]: Please correct indentation.
```suggestion
    expect = "Empty DataFrame\nColumns: ['a', 'b']\nIndex: []\n"
```

File: .gitignore (hunk 1 above)
Comment (AK-ayush) [reply .gitignore]: not needed to fix #206 . will remove it.

File: .gitignore [hunk 4]
```diff
@@ -11,7 +11,6 @@ __pycache__
 *.pytest_cache
 build
 htmlcov
-
```
Comment (harrism) [inline .gitignore]: Please remove this change (delete blank line) so that this file is not part of the PR at all.

**Comment by harrism** [comment]
Please remove .gitignore from the PR since it is unchanged.

File: .gitignore (hunk 4 above)
Comment (AK-ayush) [reply .gitignore]: sure

File: cudf/tests/test_dataframe.py (hunk 3 above)
Comment (AK-ayush) [reply cudf/tests/test_dataframe.py]: corrected indentation.

File: cudf/tests/test_dataframe.py (hunk 2 above)
Comment (AK-ayush) [reply cudf/tests/test_dataframe.py]: corrected indentation.

File: .gitignore (hunk 4 above)
Comment (AK-ayush) [reply .gitignore]: did it.  `.gitignore` is not the PR now.

**Comment by AK-ayush** [comment]
@harrism could you please check it now. build has successfully completed after the required changes.

**Comment by kkraus14** [comment]
LGTM merging, thanks @AK-ayush!

**Comment by AK-ayush** [comment]
My pleasure!
